sql/analyzer/resolve_columns.go

```diff
@@ -770,6 +772,18 @@ func indexColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (map[
 		}
 	}
 
+	// having node need access to all table references within its children node,
+	// so reset idx for any additional schema
+	switch n.(type) {
+	case *plan.Having:
+		transform.Inspect(n, func(node sql.Node) bool {
+			for _, child := range n.Children() {
+				indexChildNode(child)
+			}
+			return true
+		})
+	}
+
```

Comment:
I wanna poke on this a little bit more... it seems odd that we'd need to do this for `Having` but not for other similar nodes, like `Filter`. I'm wondering if we should be getting this information from the `availableNames` type instead of `indexColumns`. The two are pretty similar but have a couple of key differences. Let me know if you have any additional context/clues.

Reply:
Do you mean that we could use `availableNames` instead of result of `indexColumns` to resolve the columns?

The reason I added this is because `indexColumns` method gets columns in the first generation children schema of the `Having` node. In the case of `GroupBy` node as child of `Having` node, then it only gets the `SelectExprs` of `GroupBy` node as the schema, which does not include the `ResolvedTable` schema, which is a child of `GroupBy` node.